Sources/CredentialsJWT/CredentialsJWT.swift

```diff
@@ -192,90 +194,94 @@ public class CredentialsJWT<C: Claims>: CredentialsPluginProtocol {
+        noTokenType = (request.headers["X-token-type"] == nil)
+        
+        guard noTokenType || request.headers["X-token-type"] == .some(self.name) else {
+            onPass(nil, nil)
+            return
+        }
         
-                    self.usersCache?.setObject(newCacheElement, forKey: key)
-                    onSuccess(userProfile)
-                } catch {
-                    // Authorization header did not contain a valid JWT
-                    if (noTokenType) {
-                        // No X-token-type header: Allow other plugins to attempt to authenticate the Authorization header
-                        onPass(nil, nil)
-                    } else {
-                        Log.info("JWT can't be verified: \(error)")
-                        onFailure(nil, nil)
-                    }
-                }
-                
+        guard let rawToken = request.headers["Authorization"] else {
+            // No Authorization header
+            if (noTokenType) {
+                // No X-token-type header: Allow other plugins to authenticate
+                onPass(nil, nil)
             } else {
-                // No Authorization header
-                if (noTokenType) {
-                    // No X-token-type header: Allow other plugins to authenticate
-                    onPass(nil, nil)
-                } else {
-                    Log.debug("Missing authorization header")
-                    onFailure(nil, nil)
-                }
+                Log.debug("Missing authorization header")
+                onFailure(nil, nil)
             }
             
-        } else {
-            onPass(nil, nil)
+            return
+        }
+
+        let token: String
+        if rawToken.hasPrefix("Bearer") {
+            let rawTokenParts = rawToken.split(separator: " ", maxSplits: 2)
+            
+            // Added 10/19/19-- previously there was no check for the number of parts parsed, and the following array index could fail and crash the plugin/app. This would happen if the rawToken was of the form "BearerXXXXX" without a space after "Bearer".
+            guard rawTokenParts.count >= 2 else {
+                Log.debug("Badly formatted authorization header")
+                onFailure(nil, nil)
+                return
+            }
+            
+            token = String(rawTokenParts[1])
+        }
+        else {
+            token = rawToken
+        }
+        
+        getProfileAndCacheIfNeeded(token: token, options: options) { result in
+            switch result {
+            case .success(let userProfile):
+                onSuccess(userProfile)
+            case .other:
+                onPass(nil, nil)
+            case .failure(let statusCode, let dict):
+                onFailure(statusCode, dict)
+            }
         }
     }
     
+    public func generateNewProfile(token: String, options: [String : Any], completion: @escaping (CredentialsTokenTTLResult) -> Void) {
```

Comment:
Could you add a comment for this function?  It would be worth describing how it responds - ie. success if the JWT checks out, failure if the JWT can't be verified / decoded, or `other` (possibly being renamed) if the token doesn't appear to be a JWT.